tests/network/link_state.go

```diff
@@ -0,0 +1,229 @@
+
+			By("flipping the state of both interfaces")
+
+			vm, err := kubevirt.Client().VirtualMachine(vmName.Namespace).Get(context.Background(), vmName.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(PatchFlipVMInterfacesStates(vm)).To(Succeed())
+
+			vmi = WaitForStatusToShowExpectedStates(vmName, map[string]v1.InterfaceState{
+				defaultInterfaceName: v1.InterfaceStateLinkDown,
+				bridgeInterfaceName2: v1.InterfaceStateLinkUp,
+			})
+			timeout = time.Second * 30
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac1, regexLinkStateDOWN), timeout)).To(Succeed())
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac2, regexLinkStateUP), timeout)).To(Succeed())
+		})
+
+		It("hot plugging an interface with link state down and migrating vmi", func() {
+			hotplugNet := newMultusNetwork(bridgeInterfaceName3, bridgeInterfaceNAD3)
+			hotplugIface := libvmi.InterfaceWithMac(libvmi.InterfaceWithState(
+				pointer.P(libvmi.InterfaceDeviceWithBridgeBinding(bridgeInterfaceName3)),
+				v1.InterfaceStateLinkDown),
+				mac3.String())
+
+			vm, err := kubevirt.Client().VirtualMachine(testsuite.GetTestNamespace(nil)).Get(context.Background(), vmName.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(libnet.PatchVMWithNewInterface(vm, *hotplugNet, *hotplugIface)).To(Succeed())
+
+			migration := libmigration.New(vmName.Name, vmName.Namespace)
+			migrationUID := libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(kubevirt.Client(), migration)
+			libmigration.ConfirmVMIPostMigration(kubevirt.Client(), bareNamedVMI(vmName), migrationUID)
+
+			vmi := WaitForStatusToShowExpectedStates(vmName, map[string]v1.InterfaceState{
+				defaultInterfaceName: v1.InterfaceStateLinkUp,
+				bridgeInterfaceName2: v1.InterfaceStateLinkDown,
+				bridgeInterfaceName3: v1.InterfaceStateLinkDown,
+			})
+			Expect(console.LoginToFedora(vmi)).To(Succeed())
+			timeout := time.Second * 30
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac1, regexLinkStateUP), timeout)).To(Succeed())
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac2, regexLinkStateDOWN), timeout)).To(Succeed())
+			Expect(console.RunCommand(vmi, fmt.Sprintf(ipLinkTemplate, mac3, regexLinkStateDOWN), timeout)).To(Succeed())
+
+		})
+	})
+})
+
+func bareNamedVMI(vmName types.NamespacedName) *v1.VirtualMachineInstance {
+	return &v1.VirtualMachineInstance{
+		ObjectMeta: metav1.ObjectMeta{Name: vmName.Name, Namespace: vmName.Namespace},
+	}
+}
+
+func newMultusNetwork(name, netAttachDefName string) *v1.Network {
+	return &v1.Network{
+		Name: name,
+		NetworkSource: v1.NetworkSource{
+			Multus: &v1.MultusNetwork{
+				NetworkName: netAttachDefName,
+			},
+		},
+	}
+}
+
+func WaitForStatusToShowExpectedStates(name types.NamespacedName, ifaceNameToState map[string]v1.InterfaceState) *v1.VirtualMachineInstance {
```

Comment:
As this is the only file that uses `WaitForStatusToShowExpectedStates`, it could be unexported.
Please consider taking the spec interfaces slice as input, as it does already have the required info:
```suggestion
func waitForStatusToShowExpectedStates(name types.NamespacedName, specIfaces []v1.Interface) *v1.VirtualMachineInstance {
```

It would make the function call a bit shorter, and with less details:
```go
specIfaces := vm.Spec.Template.Spec.Domain.Devices.Interfaces
waitForStatusToShowExpectedStates(vmName, specIfaces)
```